nodes/bug_master

```diff
@@ -132,9 +132,20 @@ def createParametersFromLaunchFile():
         rospy.get_param('avoidance_ang_vel_magnitude'),
         rospy.get_param('avoid_distance'),
         rospy.get_param('history_cap'),
-        rospy.get_param('yaw_oscillation_tolerance')
+        rospy.get_param('yaw_oscillation_tolerance'),
+        rospy.get_param('yaw_oscillation_window'),
+        rospy.get_param('is_obstructed_window')
     )
 
+def try_set_initial_goal():
+    global state_variables_
```

Comment:
We use globals quite often in bug_master. Why don't we use locks (or some other protection mechanism) for state_variables_?